lib/util/util.js

```diff
@@ -50,6 +50,31 @@ const Util = {
     standardizedSplitChar: '/',
     skipInteraction: false,
 
+    /**
+     * helper to quickly ensure we are working with a flat Array of items
+     * @param {MetadataTypeMap|MetadataTypeItem[]} metadata map or array of metadata
+     * @returns {MetadataTypeItem[]} Array of metadata items
+     */
+    getMetadataMapToList(metadata) {
+        if (metadata && Array.isArray(metadata)) {
+            return metadata;
+        } else {
+            return Object.keys(metadata).map((item) => metadata[item]);
+        }
+    },
+    /**
+     * helper to turn an array or object into a single item
+     * @param {Object.<{metadata:MetadataTypeMap,type:string}>} obj map or array of metadata
+     * @returns {Object.<{metadata:MetadataTypeItem,type:string}>} item list reduced to single item
+     */
+    reduceMetadataMap(obj) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values
also, i wonder if the name like "extractSingleObject" wouldnt be better? reduce has a specific meaning in js, so may be better to avoid, since it isnt reducing like you might expect

Reply:
renamed to extractSingleItem()

Reply:
removed method